aptos-move/framework/aptos-framework/sources/delegation_pool.move

```diff
@@ -0,0 +1,1719 @@
+ * 12b. Synchronization detects no new inactive stake, but some pending_inactive rewards which are
+ * distributed to the (2nd) pending_inactive pool
+ * 12c. A's 50 shares = (50 * 100) / 50 = 100 coins are redeemed from the (1st) inactive pool and 100 stake
+ * is transferred to A
+ */
+module aptos_framework::delegation_pool {
+    use std::error;
+    use std::signer;
+    use std::vector;
+
+    use aptos_std::math64::min;
+    use aptos_std::pool_u64_unbound::{Self as pool_u64, total_coins};
+    use aptos_std::table::{Self, Table};
+
+    use aptos_framework::account;
+    use aptos_framework::aptos_coin::AptosCoin;
+    use aptos_framework::coin;
+    use aptos_framework::event::{Self, EventHandle};
+    use aptos_framework::stake;
+    use aptos_framework::staking_config;
+    use aptos_framework::timestamp;
+
+    const MODULE_SALT: vector<u8> = b"aptos_framework::delegation_pool";
+
+    const VALIDATOR_STATUS_INACTIVE: u64 = 4;
+
+    /// Delegation pool owner capability does not exist at the provided account.
+    const EOWNER_CAP_NOT_FOUND: u64 = 1;
+
+    /// Account is already owning a delegation pool.
+    const EOWNER_CAP_ALREADY_EXISTS: u64 = 2;
+
+    /// Delegation pool does not exist at the provided pool address.
+    const EDELEGATION_POOL_DOES_NOT_EXIST: u64 = 3;
+
+    /// There is a pending withdrawal to be executed before unlocking any stake
+    const EPENDING_WITHDRAWAL_EXISTS: u64 = 4;
+
+    /// Commission percentage has to be between 0 and `MAX_FEE` - 100%.
+    const EINVALID_COMMISSION_PERCENTAGE: u64 = 5;
+
+    /// Slashing (if implemented) should not be applied to already `inactive` stake.
+    /// Not only it invalidates the accounting of past observed lockup cycles (OLC),
+    /// but is also unfair to delegators whose stake has been inactive before validator started misbehaving.
+    /// Additionally, the inactive stake does not count on the voting power of validator.
+    const ESLASHED_INACTIVE_STAKE_ON_PAST_OLC: u64 = 6;
+
+    const MAX_U64: u64 = 18446744073709551615;
+
+    /// Maximum operator percentage fee(of double digit precision): 22.85% is represented as 2285
+    const MAX_FEE: u64 = 10000;
+
+    /// Capability that represents ownership over privileged operations on the underlying stake pool.
+    struct DelegationPoolOwnership has key, store {
+        /// equal to address of the resource account owning the stake pool
+        pool_address: address,
+    }
+
+    struct DelegationPool has key {
+        // Share pool of `active` + `pending_active` stake
+        active_shares: pool_u64::Pool,
+        // Index of current lockup cycle on the delegation pool since its creation
+        observed_lockup_cycle: u64,
+        // Share pools of `inactive` stake on each past lockup cycle and `pending_inactive` stake on current one
+        inactive_shares: Table<u64, pool_u64::Pool>,
+        // Unique lockup cycle (key in `inactive_shares`) where delegator has stake to withdraw
```

Comment:
Can you update this to
`Mapping from delegator address to observed lockup cycle if they have any pending withdrawal.`